### DISCUSSION TITLE
Converting new NixOS-WSL install to a flake results in building mesa from sources

### DISCUSSION THREAD

**Discussion by alexvorobiev** [nix-community/NixOS-WSL#265] [Q&A] [answered]
I am trying to switch a new instance of NxOS-WSL to a minimal flake (basically just added this repo to the inputs, `configuraion.nix` to modules, and removed the corresponding import statement from configuration.nix). `nixos-rebuild switch` doesn't show any errors but for some reason it now wants to build mesa from scratch. I tried both unstable and 23.05. I don't even understand why it needs mesa in a barebones text-only configuration:

```
# nixos-rebuild dry-build
building the system configuration...
these 9 derivations will be built:
  /nix/store/jcc3wzxqqs3ri7w3rzgg8mxqiap8p1i4-nixos-wsl-utils-1.0.0.drv
  /nix/store/npazw3k850dmkqlmzjmwy1g44fxpcs86-set-environment.drv
  /nix/store/canx5r3xgmm0rhr1j1dv3vj6k4w6wlcv-etc-profile.drv
  /nix/store/iixyw307kb3bhl5ahvln4c5s9fv3snzl-mesa-23.0.3.drv
  /nix/store/pi7sagaqv02m3bqqyxd4czhj74p3g7zv-opengl-drivers.drv
  /nix/store/m4wfv153r9cwi1nz979whaq42pxg1lcq-nixos-tmpfiles.d.drv
  /nix/store/grmlkx6xfwngy760q2lpcydbyxr245pb-tmpfiles.d.drv
  /nix/store/rmsaq6ay97xaijqbydisk2vgkh3nbmd5-etc.drv
  /nix/store/rpkqy6a9sybr2asp7hkxs9nipznix3x4-nixos-system-cheeseshop-23.05.20230707.d8bb6c6.drv
```

**Comment by SuperSandro2000** [comment]
Because if you want to use wslg, you probably want to have hardware acceleration https://github.com/nix-community/NixOS-WSL/blob/0982e9ab209aee459ed3331ab4eadbb4d8a023e1/modules/wsl-distro.nix#L71C11-L71C19

No idea why that isn't cached for you though.

**Comment by alexvorobiev** [comment, marked as answer]
It looks like I found the culprit - my `configuration.nix` had these lines
```
imports = [
    "${modulesPath}/profiles/minimal.nix"
 ];
```
which enables `environment.noXlibs` with a disastrous effect on pretty much everything that needs X11. I think the result of putting those lines to the default `configuration.nix` needs to be explained somewhere in the documentation. It looks I am not the only one who was bitten by that: https://github.com/nix-community/NixOS-WSL/issues/33.

**Reply by SuperSandro2000** [reply]
That is no longer included in the next stable verison/on master.
